Feature/auto join channels

AaronFoley opened the pull request:
- Switched to using the MessageChannelEvent.Chat event rather than Player.setMessageChannel as this sends all messages, even non-chat messages generated by the player e.g. Death/Join

- Implemented a simple command to send a message directly to a channel without switching your default channel

- Implemented an option to specify a list of channels for players to automatically join #1

AaronFoley committed:
Marked Gradlew as executable

AaronFoley committed:
Added support for Auto-Join Channels and Added a command to speak directly to a channel

LJNIC commented:
As I mentioned already, you should use `@Root Player player` for the message event. Also, I'm not sure about the change to `getPlayerChannel`. The intention was for it to get any channel the player was in, and set their new channel to a channel they were already in, instead of just the default channel.

AaronFoley committed:
Refactor speaking map

AaronFoley commented:
Reverted the logic in `getPlayerChannel` and added a new method with the new logic `getPlayerSpeakingChannel`

Also noticed that the facade was basically setting the speaking channel when leaving a channel so I've moved the logic out of the service layer and into the facade.
